includes/AdvertiseASC/NewBuyers.php

```diff
@@ -0,0 +1,306 @@
+		}
+		
+		return $properties;
+	}
+
+	private function get_ad_creation_params( $ad_name, $adset_id, $creative ) {
+
+		return array(
+			'name'     => $ad_name,
+			'adset_id' => $adset_id,
+			'creative' => $creative,
+		);
+
+	}
+
+	private function setup_ad( $adset_id, $ad_name, $page_id, $ad_description_message, $product_set_id, $store_url ) {
+
+		$creative = $this->get_adcreative_creation_params( $this->ad_creative_name, $page_id, $store_url, $ad_description_message, $product_set_id );
+		return $this->create_ad( $this->get_ad_creation_params( $ad_name, $adset_id, $creative ) );
+
+	}
+
+	public function create_asc_campaign( $props ) {
+
+		$status = $props['state'] == 'true' ? 1 : 0;
+		$daily_budget = $props['daily_budget'];
+		$ad_message = $props['ad_message'];
+		$countryList = $props['country'];
+
+		$this->facebook_pixel_id = $this->integration->get_facebook_pixel_id();
+
+		try {
+
+			$this->campaign = $this->setup_campaign();
+
+		} catch ( ApiException $e ) {
+
+			if ( $e->getCode() === 2 ) {
+
+				throw new AscNotSupportedException();
+
+			} else {
+
+				$message = sprintf( 'An exception happened trying to setup the New Buyers campaign for the first time. ' . $e->getMessage() );
+				\WC_Facebookcommerce_Utils::log( $message );
+				throw new PluginException( $message );
+
+			}
+		}
+
+		try {
+
+			$this->adset      = $this->setup_adset( $this->facebook_pixel_id, $this->campaign['id'], $daily_budget * 100, $countryList);
+			$this->ad         = $this->setup_ad( $this->adset['id'], $this->ad_name, $this->facebook_page_id, $ad_message, $this->default_product_set, $this->store_url );
+			$this->adcreative = $this->fetch_adcreative( $this->ad['adcreatives']['data'][0]['id'] );
+
+		} catch ( ApiException $e ) {
+
+			$message = sprintf( 'An exception happened trying to setup the New Buyers campaign objects for the first time. ' . $e->getMessage() );
+			\WC_Facebookcommerce_Utils::log( $message );
+			if ( str_contains( $e->getMessage(), 'non-discrimination' ) ) {
+				throw new NonDiscriminationNotAcceptedException();
+			} else {
+				throw new PluginException( $message );
+			}
+		}
+
+		$this->update_stored_data();
+
+		if ($status) {
+			$this->set_ad_status($status);
+		}
+
+		$this->get_insights();
+	}
+
+	public function update_asc_campaign( $data ) {
+
+		// if ( ! $data ) {
+		// 	return true;
+		// }
+
+		// $msg = array_key_exists( 'p2', $data ) ? $data['p2'] : '';
+
+		// if ( $msg ) {
+		// 	$this->apply_creative_changes( $msg );
+		// }
+
+		// $countries    = array_key_exists( 'p4', $data ) ? explode( ',', $data['p4'] ) : [];
+		// $daily_budget = array_key_exists( 'p5', $data ) ? $data['p5'] : '';
+
+		// if ( $countries || $daily_budget ) {
+		// 	$this->apply_adset_changes( $countries, $daily_budget );
+		// }
+
+		// if ( array_key_exists( 'p1', $data ) ) {
+		// 	$this->set_ad_status( $data['p1'] );
+		// }
+
+		// return true;
```

Comment:
We'll need to clean up the commented out codes. This method isn't used anywhere it seems